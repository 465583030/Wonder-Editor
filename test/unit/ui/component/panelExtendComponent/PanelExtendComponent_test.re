open Wonder_jest;

open Expect;

open Expect.Operators;

open Sinon;

let _ =
  describe(
    "panel extension component",
    (_) => {
      test(
        "accord to user json data, build component",
        (_) => {
          let component =
            ReactTestRenderer.create(
              ExtensionTool.buildSpecificExtesion("App", ExtensionTool.extensionText, 0)
            );
          let json = ReactTestRenderer.toJSON(component);
          toMatchSnapshot(expect(json))
        }
      );
      test(
        "the extensionText specific parent is App,if not, don't render",
        (_) => {
          let component =
            ReactTestRenderer.create(
              ExtensionTool.buildSpecificExtesion("MainEditor", ExtensionTool.extensionText, 0)
            );
          let json = ReactTestRenderer.toJSON(component);
          toMatchSnapshot(expect(json))
        }
      )
    }
  );