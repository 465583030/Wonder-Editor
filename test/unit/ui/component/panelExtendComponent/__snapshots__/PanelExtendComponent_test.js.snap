// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`panel extension component accord to user json data, build component 1`] = `
<div>
  <div
    className="inline-component"
  >
    <button>
      xme
    </button>
  </div>
  <div
    className="inline-component"
  >
    <div
      className="number-input"
    >
      <span
        className="number-label"
      >
        xXX : 
      </span>
      <input
        className="ant-input number-input-input"
        onChange={[Function]}
        type="text"
        value="0"
      />
    </div>
  </div>
</div>
`;

exports[`panel extension component the extensionText specific parent is App,if not, don't render 1`] = `
<div
  className="fff"
/>
`;
